doc/specs/bookmarks.md

```diff
@@ -0,0 +1,595 @@
+# Bookmarks
+
+## Metadata
+
+Author: killasnake0-wcd
+
+
+## Table of contents
+
+1. Motivation
+1. Business rules
+1. Use cases
+
+## Motivation
+
+Provide a system for users to bookmark torrent groups, artists, collages, 
+or requests made on the website
+
+## Business rules
+
+* Users are able to bookmark a torrent group, artist, collage, or request.
+* Users are able to remove bookmarks
+* Users are able to sort bookmarks in ascending or descending order based on different fields.
+* Users can receive notifications about certain bookmarks. 
+
+## Use cases
+
+### 1. Torrent Group Bookmarking
+
+#### 1.0 Viewing All Bookmarked Torrent Groups
+
+**Primary Actor:** Any user.
+
+**Brief:** Allows the user to view their torrent group bookmarks.
+
+**Trigger:** The user wants to view torrent group bookmarks.
+
+**Preconditions:** The user has already bookmarked at least one torrent group. 
+
+**Basic Flow:**
+ 1. The user navigates to the Bookmarks page.
+ 2. The user clicks on the Torrents hyperlink.
+
+**Additional non-functional requirements:** 
+ - There is a cover art section. 
+ - Each "row" for the torrent group should appear as it would on the artist page
+
+ 
+#### 1.1 Adding a Torrent Group Bookmark
+
+**Primary Actor:** Any user.
+
+**Brief:** Allows a user to add a bookmark for an torrent group.
+
+**Trigger:** The user wishes to bookmark a torrent group.
+
+**Preconditions:** The user has not already bookmarked the torrent group
+
+**Postconditions:** The torrent group is now bookmarked for that user
+
+**Basic Flow:**
+
+ 1. The user navigates to the torrent group page.
+ 2. The user clicks the "Bookmark" hyperlink.
+ 3. The torrent group is bookmarked for that user.
+ 
+**Additional functional requirements:**
+ - The hyperlink function should also change to that of removing the bookmark
+
+**Additional non-functional requirements:** 
+ -  The hyperlink text is updated to "Remove Bookmark"
+ 
+ 
+#### 1.2 Removing a Torrent Group Bookmark
+
+**Primary Actor:** Any user.
+
+**Brief:** Allows a user to remove a bookmark for a torrent group.
+
+**Trigger:** The user wishes to remove the bookmark for a torrent group.
+
+**Preconditions:** The user has already bookmarked the torrent group.
+
+**Postconditions:** The torrent group is no longer bookmarked for that user. Also the 
+torrent ranking will be spliced from the overall bookmark list.
+
+**Basic Flow:** 
+
+ 1. The user navigates to the torrent group page.
+ 2. The user clicks the "Remove Bookmark" hyperlink.
```

Comment:
This is too detailed and may not reflect how the page actually works. Just describe what needs to happen, not what the user presses. How do you know it's a hyperlink anyways?